broker-daemon/block-order-worker/index.js

```diff
@@ -892,13 +895,12 @@ class BlockOrderWorker extends EventEmitter {
    * @private
    * @returns {Promise<boolean>}
    */
-  async relayerIsAvailable () {
-    try {
-      await this.relayer.adminService.healthCheck({})
-      return true
-    } catch (e) {
-      return false
+  relayerIsAvailable () {
+    if (!this.cachedCall) {
+      this.cachedCall = new CachedCall(() => this.relayer.adminService.healthCheck({}))
```

Comment:
@bradleystachurski suggestion for how to handle `CachedCall` without needing an anonymous function.

Have you tried having 2 arguments to CachedCall:

```
func = this.relayer.adminService.healthCheck
params = {}
new CachedCall(func, params)
```

then inside of the CachedCall class

```
return this.lastCallFn(this.lastCallParams)
```

Reply:
Capturing our in-person chat. Going to keep the anonymous function here since the fat arrow keeps the context of `this` without being overly verbose with `Function.prototype.bind`.